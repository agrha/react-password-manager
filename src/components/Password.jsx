import React, { Component } from 'react';
import {Jumbotron, Button, Thumbnail} from 'react-bootstrap'
import {BrowserRouter as Router, Link} from 'react-router-dom'
import './Restaurant.css';

class Restaurant extends Component {

  refresh () {
    window.location.href(`/${this.props.data.restaurant.R.res_id}/${this.props.category}`)
  }
    
  render() {
    return (
      <Router>
        <Jumbotron>
            <Thumbnail src={this.props.data.restaurant.featured_image || 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAMFBMVEXy8vK8vLz19fW5ubm2trbl5eXZ2dni4uLDw8PJycnW1tbr6+vNzc3R0dHp6ene3t6up7FsAAAFqklEQVR4nO2bi5Ksqg6GJUFugrz/254E8NKzp3p6T3fXHHv/X82yFDHwQxJsdU0TAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC8C6I/bvtL+/Tt7jOk8ho7Bw93jIqjqd627/Oy7a5lfUl3DL/CzIk1z4+2bQNRscu5yNldlrf1Ff0hY15h5sS/6Fj1Mo/8ReF++FKFpH/UAmP4GB0RSrQX687hhP1oFGwhRd76U5Ut2A6zJ9t6dVdINK4WhdvpofDo0xMK1+DnYIInZ0yYtYU1GjYtQmiJxqQpej0oct5t7ZELlKSgtoLcL6AihsJmvZmRyKIc2tT4ILPmg+GgtikWGgqT1FNDqrDIabXZFS7ZaM+eVWgjZzGcOWfWFldrksus07taOcXBirBFaqRo4zYfhaMpxViNO8PFFbmAUuSYh/FFzfRSvV4k2YmqDc5FCUGJw9gVSnl2Sdsgx8GUZGyirnBlLnKm/F5iV8i8ElURJ1ttJ4uvkHRLot4Y2RUBUhq1lqSGuisM4kPeZr1gprF7eKlmESl1UrCwKJoWFkmB1e2zCtsULqJAG4lN4dTaU3evUlubF/O/T6tjDjMdWxkv77UXVbo962hOTXfvB008JlEEqEe3414mvW8d24zPlZpucc2oikSrpBcdi5ZCD4VVZZAJfThaONam0Ot4U2v6SYWJjm3R7DHXJJE1S1Mt9y/SUuVcnXPVhF1hCy4Z+rk5YZ+0k8LNjO9DJBPSEop3JfJ5DiXYpSxwU9hsal9UYeKibbptVH+v0N0opNlYGyT85u6pXaFj0xmJRPRMQ6F25huFq5gxYkazlPSxzYQMlJVEE88KxQutCanP4dTb6wrL1uaLFU5s5uGlYw61RrXN6YiOMNsUbnOY7Y2XGjMPL22n2vwsNixfvFSGsUy7l7Y5VINtDlt8H22+SuEIPo0z33eTZJp1BENMu5duCpeeYMUYnRbqfoH00rc+16D5s7bppvMcykGr37209mvWEYetsSX+fvH/VqHmHHFVFi/tuVR2NYxGLvXTV4WS7n1Lvq4pHAOuuokkUbfkYQK7tsql5h63CsXwlJtC1vZa6m2GWvNi/plcyt0nbuJQlqsSbLKy1EoHg6xHPNbDcqxNm0LtzWJslFM6k7IIboE6zHDwbVrGgLDUZFlL165QQne2uhBHWZRlDrONLQ91Z5CxlZXapieW/JSkU0mDbUl+35YQisRHnCX1uVzmpQea3LDE4/7Cp26hjlP9dodq3n8upBCymumGx9yqbdlGOVG7EZpjiFVupDLNZXLattRb06pZNocQ52fu206/0U7bHtv6b522NW0vPq49LEznFLTXOcycio9C2o2cy3ZDdKr+Phb9haMh+M5G/hSJk6D3nvWt4/inUI0m5PVzBU7vDwQAAAAAAPBG1vkhlove8dFq+EHC8rO5/0OW7Snfz3C44ixSEoXxIYxh/9fd/QWkD7LpIRKzu+AkNoWP1by6QprW5e7v48srLJbZ3HuGc3GFlFtKvfeY6uIK57Fm3HnUeG2F+7svvvm+hM5Hn6LwnFfbu+7j6NoKt3em509iyLPhQ+K1FU5TV3h+Ba0CzUnixRXqe0W98Tyd8EP0JvHiCidaUozuVO63O/JN4tUVfnnkfwjcJV5f4U2pP/+m6hI/SuGtwCHxMxSG9mb/q8Au8SMUBmb9EugfAkVi/Yjfh0EXjPKdwPa53/UVBu4O+d2Dm49QGO49kvoEhXcFfoLC+wKvr/D4NvJjFX6bXqDwGvxnFJbwA/Xqd20/PtK/+J33Q/2+rMLCnJaHiOaSCqfKj74hNcZe8hUpxYdfkfIz32j/IeTiT4m0E5/5f2d/ymPvR/FpKgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAODz+B9bJTiKWY5GtAAAAABJRU5ErkJggg==' } alt="242x200"/>
            <h1>{this.props.data.restaurant.name}</h1>
            <p className="restaurant-name">Price for Two: {this.props.data.restaurant.average_cost_for_two} {this.props.data.restaurant.currency}</p>
            <p>{this.props.data.restaurant.location.address}</p>
            <p>Restauranteer Votes: {this.props.data.restaurant.user_rating.votes}</p>
            <Link to={`/${this.props.data.restaurant.R.res_id}/${this.props.category}`} onClick={this.refresh}><Button bsStyle="primary">See Detail</Button></Link>
        </Jumbotron>
      </Router>
    );
  }
}

export default Restaurant;
