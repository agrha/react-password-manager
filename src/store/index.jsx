import {createStore, combineReducers, applyMiddleware} from 'redux'
import passwordReducer from './password/reducer.password'
import thunk from 'redux-thunk'

const reducer = combineReducers({
  password: passwordReducer
})

const store = createStore (
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  applyMiddleware(thunk)
)

export default store