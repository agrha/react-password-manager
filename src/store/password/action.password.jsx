import password from '../../firebase'

import {
  LOADING_PASSWORDS,
  SUCCESS_LOAD_PASSWORDS,
  ERROR_LOAD_PASSWORDS,
} from './actionTypes.password'

export function getAllPassword () {
  return dispatch => {
    dispatch(loadingPasswords())
    let passwordList = []
    password.on('value', snapshot => {
      let pw = snapshot.val()
      for(let key in pw) {
        if(pw.hasOwnProperty(key)) {
          passwordList.push({key, ...pw[key]})
        }
      }
      console.log(passwordList, 'ini array di get All')
      dispatch(successLoadPassword(passwordList))
    }, err => {
      dispatch(errorLoadPasswords())
    })
  }
}

export function loadingPasswords () {
  return {
    type: LOADING_PASSWORDS
  }
}

export function successLoadPassword (payload) {
  return {
    type: SUCCESS_LOAD_PASSWORDS,
    payload: payload
  }
}

export function errorLoadPasswords () {
  return {
    type: ERROR_LOAD_PASSWORDS
  }
}

export function SuccessAddPassword (payload) {
  return {
    type: ADD_PASSWORD,
    payload: payload
  }
}

export function deletePassword () {
  return {
    type: DELETE_PASSWORD
  }
}

export function updatePassword (payload) {
  return {
    type: UPDATE_PASSWORD,
    payload: payload
  }
}