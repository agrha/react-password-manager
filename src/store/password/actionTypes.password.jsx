export const LOADING_PASSWORDS = 'LOADING_PASSWORD'
export const SUCCESS_LOAD_PASSWORDS = 'SUCCESS_LOAD_PASSWORDS'
export const ERROR_LOAD_PASSWORDS = 'ERROR_LOAD_PASSWRODS'

export const ADD_PASSWORD = 'ADD_PASSWORD'
export const DELETE_PASSWORD = 'DELETE_PASSWORD'
export const UPDATE_PASSWORD = 'UPDATE_PASSWORD'