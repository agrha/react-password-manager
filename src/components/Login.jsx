import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { getAllUsers } from '../store/user/action.user'
import { Link } from 'react-router-dom'
import './Password.css'

class Login extends Component {
  constructor() {
    super()
    this.state = {
      key:'',
      username: '',
      password: '',
    }
  }

  UNSAFE_componentWillMount () {
    this.props.getAllUsers()
  }

  checkLength () {
    if (this.state.password.length > 5) {
      this.setState({
        length: true
      })
    } else {
      this.setState({
        length: false
      })
    }
  }

  checkUpper () {
    let regexUpper = /[A-Z]/

    if (regexUpper.test(this.state.password)) {
      this.setState({
        upperCase: true
      })
    } else {
      this.setState({
        upperCase: false
      })
    }
  }

  checkLower () {
    let regexLower = /[a-z]/

    if (regexLower.test(this.state.password)) {
      this.setState({
        lowerCase: true
      })
    } else {
      this.setState({
        lowerCase: false
      })
    }
  }

  checkNumber () {
    let regexNumber = /[0-9]/

    if (regexNumber.test(this.state.password)) {
      this.setState({
        numberCase: true
      })
    } else {
      this.setState({
        numberCase: false
      })
    }
  }

  checkSpecial () {
    let regexSpesial = /[^A-Za-z0-9]/
    
    if (regexSpesial.test(this.state.password)) {
      this.setState({
        specialCase: true
      })
    } else {
      this.setState({
        specialCase: false
      })
    }
  }

  handleOnChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    }, () => {
      this.checkLength()
      this.checkUpper()
      this.checkLower()
      this.checkNumber()
      this.checkSpecial()
    })
  }

  login = (e) => {
    e.preventDefault()
    if (
      this.state.specialCase &&
      this.state.upperCase &&
      this.state.lowerCase &&
      this.state.length &&
      this.state.numberCase &&
      this.state.username !== '' &&
      this.state.password !== ''
    ) {
        let userLogin = {
          username: this.state.username,
          password: this.state.password,
          createdAt: this.state.createdAt,
        }
        this.props.users.map(user => {
          if(user.username === userLogin.username){
            if(user.password === userLogin.password){
              localStorage.setItem('userKey', user.key)
              localStorage.setItem('username', user.username)
              localStorage.setItem('password', user.password)
              this.props.history.push('/home')
            } else {
              alert('wrong password')
            }
          }
        })
      }
  }

  render () {
    return (
      <div className="container">
        <div className="flex-frame">
          <form onSubmit={this.login}>
            <fieldset>
              <h1><strong>Login Form</strong></h1>
                <div className="form-group">
                  <label htmlFor="inputUsername">Username</label>
                  <input
                    type="text"
                    className="form-control"
                    id="inputUsername"
                    placeholder="Enter Your Username"
                    name="username"
                    value={this.state.username}
                    onChange={this.handleOnChange}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="inputPass">Password</label>
                  <input
                    type="password"
                    className="form-control"
                    id="inputPass"
                    placeholder="Enter Your Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.handleOnChange}
                    autoFocus
                  />
                </div>
                <div className="form-group bold">
                  <div className="custom-control custom-checkbox">
                    <input className="custom-control-input" id="checkUpper" type="checkbox" checked={this.state.upperCase} />
                    <label className="custom-control-label" id="labelUpper" htmlFor="checkUpper">at least one uppercase letter?</label>
                  </div>
                  <div className="custom-control custom-checkbox">
                    <input className="custom-control-input" id="checkLower" type="checkbox" checked={this.state.lowerCase} />
                    <label className="custom-control-label" id="labelLower" htmlFor="checkLower">at least one lowercase letter? </label>
                  </div>
                  <div className="custom-control custom-checkbox">
                    <input className="custom-control-input" id="checkSpecial" type="checkbox" checked={this.state.specialCase} />
                    <label className="custom-control-label" id="labelSpecial" htmlFor="checkSpecial">at least one symbol?</label>
                  </div>
                  <div className="custom-control custom-checkbox">
                    <input className="custom-control-input" id="checkNumber" type="checkbox" checked={this.state.numberCase} />
                    <label className="custom-control-label" id="labelNumber" htmlFor="checkNumber">at least one number?</label>
                  </div>
                  <div className="custom-control custom-checkbox">
                    <input className="custom-control-input" id="checkLength" type="checkbox" checked={this.state.length} />
                    <label className="custom-control-label" id="labelLength" htmlFor="checkLength">minimal 5 letter length?</label>
                  </div>
                </div>
              <button disabled={!(
                this.state.specialCase &&
                this.state.upperCase &&
                this.state.lowerCase &&
                this.state.length &&
                this.state.numberCase &&
                this.state.username !== '' &&
                this.state.password !== ''
              )} type="submit" className="btn btn-primary button">Submit</button>
              <Link to={`/register`}>
                <button className="btn btn-primary button">
                  To Register
                </button>
              </Link>
            </fieldset>
          </form>
        </div>
      </div>  
    )
  }
}

const mapStateToProps = (state) => ({
  users: state.user.users,
  loading: state.user.loading,
  error: state.user.error
})

const mapDispatchToProps = dispatch => bindActionCreators({
  getAllUsers
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(Login)