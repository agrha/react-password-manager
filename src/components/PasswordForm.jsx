import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { addPassword } from '../store/password/action.password'
import './Password.css'

class PasswordForm extends Component {
  constructor() {
    super()
    this.state = {
      url: '',
      username: '',
      password: '',
      upperCase: false,
      lowerCase: false,
      specialCase: false,
      length: false,
      numberCase: false
    }
  }

  checkLength () {
    if (this.state.password.length > 5) {
      this.setState({
        length: true
      })
    } else {
      this.setState({
        length: false
      })
    }
  }

  checkUpper () {
    let regexUpper = /[A-Z]/

    if (regexUpper.test(this.state.password)) {
      this.setState({
        upperCase: true
      })
    } else {
      this.setState({
        upperCase: false
      })
    }
  }

  checkLower () {
    let regexLower = /[a-z]/

    if (regexLower.test(this.state.password)) {
      this.setState({
        lowerCase: true
      })
    } else {
      this.setState({
        lowerCase: false
      })
    }
  }

  checkNumber () {
    let regexNumber = /[0-9]/

    if (regexNumber.test(this.state.password)) {
      this.setState({
        numberCase: true
      })
    } else {
      this.setState({
        numberCase: false
      })
    }
  }

  checkSpecial () {
    let regexSpesial = /[^A-Za-z0-9]/
    
    if (regexSpesial.test(this.state.password)) {
      this.setState({
        specialCase: true
      })
    } else {
      this.setState({
        specialCase: false
      })
    }
  }

  checkClass () {
    if (
      this.state.specialCase &&
      this.state.upperCase &&
      this.state.lowerCase &&
      this.state.length &&
      this.state.numberCase &&
      this.state.password !== ''
    ) {
      this.setState({
        passClass: "form-control is-valid"
      })
    } else {
      this.setState({
        passClass: "form-control is-invalid"
      })
    }
  }

  handleOnChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    }, () => {
      this.checkClass()
      this.checkLength()
      this.checkUpper()
      this.checkLower()
      this.checkNumber()
      this.checkSpecial()
    })

  }

  handleSubmit = (e) => {
    e.preventDefault()
    if (
      this.state.specialCase &&
      this.state.upperCase &&
      this.state.lowerCase &&
      this.state.length &&
      this.state.numberCase &&
      this.state.username !== '' &&
      this.state.password !== ''
    ) {
      let date = new Date()
      let formPw = {
        url: this.state.url,
        username: this.state.username,
        password: this.state.password,
        createdAt: date.toLocaleString(),
        updatedAt: '-'
      }

      this.props.addPassword(formPw)
      this.clearForm()
    }
  }

  clearForm() {
    this.setState({
      url: '',
      username: '',
      password: '',
    })
  }

  render () {
    return (
      <div className="container col-md-5">
        <form onSubmit={this.handleSubmit}>
          <fieldset>
            <h1><strong>Password Form</strong></h1>
              <div className="form-group">
                <label htmlFor="inputUrl">URL</label>
                <input
                  type="text"
                  className="form-control"
                  id="inputUrl"
                  placeholder="Enter Your URL"
                  name="url"
                  value={this.state.url}
                  onChange={this.handleOnChange}
                />
              </div>
              <div className="form-group">
                <label htmlFor="inputUsername">Username</label>
                <input
                  type="text"
                  className="form-control"
                  id="inputUsername"
                  placeholder="Enter Your Username"
                  name="username"
                  value={this.state.username}
                  onChange={this.handleOnChange}
                />
              </div>
              <div className="form-group">
                <label htmlFor="inputPass">Password</label>
                <input
                  type="password"
                  className={this.state.passClass}
                  id="inputPass"
                  placeholder="Enter Your Password"
                  name="password"
                  value={this.state.password}
                  onChange={this.handleOnChange}
                  autoFocus
                />
              </div>
              <div className="form-group bold">
                <div className="custom-control custom-checkbox">
                  <input className="custom-control-input" id="checkUpper" type="checkbox" checked={this.state.upperCase} />
                  <label className="custom-control-label" id="labelUpper" htmlFor="checkUpper">at least one uppercase letter?</label>
                </div>
                <div className="custom-control custom-checkbox">
                  <input className="custom-control-input" id="checkLower" type="checkbox" checked={this.state.lowerCase} />
                  <label className="custom-control-label" id="labelLower" htmlFor="checkLower">at least one lowercase letter? </label>
                </div>
                <div className="custom-control custom-checkbox">
                  <input className="custom-control-input" id="checkSpecial" type="checkbox" checked={this.state.specialCase} />
                  <label className="custom-control-label" id="labelSpecial" htmlFor="checkSpecial">at least one symbol?</label>
                </div>
                <div className="custom-control custom-checkbox">
                  <input className="custom-control-input" id="checkNumber" type="checkbox" checked={this.state.numberCase} />
                  <label className="custom-control-label" id="labelNumber" htmlFor="checkNumber">at least one number?</label>
                </div>
                <div className="custom-control custom-checkbox">
                  <input className="custom-control-input" id="checkLength" type="checkbox" checked={this.state.length} />
                  <label className="custom-control-label" id="labelLength" htmlFor="checkLength">minimal 5 letter length?</label>
                </div>
              </div>
            <button disabled={!(
              this.state.specialCase &&
              this.state.upperCase &&
              this.state.lowerCase &&
              this.state.length &&
              this.state.numberCase &&
              this.state.username !== '' &&
              this.state.password !== ''
            )} type="submit" className="btn btn-primary">Submit</button>
          </fieldset>
        </form>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({
  addPassword
}, dispatch)

export default connect(null, mapDispatchToProps)(PasswordForm)