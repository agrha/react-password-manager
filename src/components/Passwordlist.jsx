import React, {Component} from 'react'
import Password from './Password.jsx'
import PasswordForm from './PasswordForm.jsx'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getAllPassword } from '../store/password/action.password'
import {BrowserRouter as Router} from 'react-router-dom'
import './Password.css'


class Passwordlist extends Component {

  UNSAFE_componentWillMount(){
    this.props.getAllPassword()
    // console.log(this.props.passwords)
  }

  render () {
    var password = this.props.passwords.map((password, index) => 
    <Password data={password} key={index}/>
    )
    return (
    <Router>
      <div className="container">
      <PasswordForm className='flex-frame'></PasswordForm>
      <table className="table table-striped">
        <thead className="thead-dark">
          <tr>
            <th scope="col">Website</th>
            <th scope="col">Username</th>
            <th scope="col">Password</th>
            <th scope="col">Action</th>
          </tr>
        </thead>
        <tbody>
          {password}
        </tbody>
      </table>
      </div> 
    </Router>
    )
  }
}

const mapStateToProps = (state) => ({
  passwords: state.password.passwords
})

const mapDispatchToProps = (dispatch) => bindActionCreators({
  getAllPassword
}, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Passwordlist);