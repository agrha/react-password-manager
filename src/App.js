import React, { Component } from 'react';
import logo from './logo.svg';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
import Passwordlist from './components/Passwordlist.jsx'
import Error from './components/Error.jsx'
import './App.css';

class App extends Component {
  render() {
    return (
    <Router>
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Switch>
          <Route exact path="/" component={Passwordlist}/>
          <Route component={Error}/>
        </Switch>
      </div>
    </Router>      
    );
  }
}

export default App;
