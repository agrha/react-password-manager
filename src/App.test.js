import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { 
  BrowserRouter as Router, Route, 
  Redirect, Switch, MemoryRouter 
} from 'react-router-dom'
import renderer from 'react-test-renderer'
import * as pwAction from './store/password/action.password'

import App from './App';
import Login from './components/Login'
import EditForm from './components/EditForm'
import Passwordlist from './components/Passwordlist'
import PasswordForm from './components/PasswordForm'
import Password from './components/Password'
import Register from './components/Register'
import store from './store'
import {getAllPassword} from './store/password/action.password'

Enzyme.configure({ adapter: new Adapter()})

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const wrapper = mount(
      <Provider store={store}>
        <App user={{loginStatus: false}}/>
      </Provider>
    )
    ReactDOM.render(wrapper, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('should have <Router />, <Switch /> and <Route />', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.containsAllMatchingElements([
      <Router />,
      <Switch />,
      <Route />
    ])).toHaveLength[5]
  })

  it('the first page should be Login', () => {
    const wrapper = mount(
      <Provider store={store}>
        <App user={{loginStatus: false}}/>
      </Provider>
    )
    expect(wrapper.containsMatchingElement(<Login />)).toBeTruthy()
    expect(wrapper.containsMatchingElement(<Passwordlist />)).toBeFalsy()
    // expect(wrapper.find('h2')).toBeTruthy()
  })
})

describe('store', () => {
  const wrapperStore = mount(
    <Provider store={store}>
      <Passwordlist/>
    </Provider>
  )
  it('passwordlist should mount data at start', () => {
    expect(wrapperStore.props().store.getState().password).toEqual({
      passwords: [],
      loading: true,
      error: false
    })
  })
})


describe('form password checked', () => {
  const wrapperForm = shallow(<PasswordForm/>, {context:{store}}).dive()
  it('PasswordForm should have form', () => {
    expect(wrapperForm.find('form')).toBeTruthy()
  })
  it('PasswordForm should have input', () => {
    expect(wrapperForm.find('input')).toHaveLength(8)
  })
  it('PasswordForm should have button Submit', () => {
    expect(wrapperForm.find('button').text()).toEqual('Submit')
  })
  it('fail check uppercase', () => {
    let pass = 'agrhaganteng'
    wrapperForm.find('input').at(3).simulate('change', {
      target: {
        name: 'password',
        value: pass
      }
    })
    wrapperForm.find('form').at(0).simulate('submit', {preventDefault() {}})
    expect(wrapperForm.find('#labelUpper').at(0).text()).toMatch("at least one uppercase letter?")
  })
  it('fail check lowercase', () => {
    let pass = 'AGRHA123'
    wrapperForm.find('input').at(3).simulate('change', {
      target: {
        name: 'password',
        value: pass
      }
    })
    wrapperForm.find('form').at(0).simulate('submit', {preventDefault() {}})
    expect(wrapperForm.find('#labelLower').at(0).text()).toMatch("at least one lowercase letter?")
  })
  it('fail check number', () => {
    let pass = 'agrhaganteng'
    wrapperForm.find('input').at(3).simulate('change', {
      target: {
        name: 'password',
        value: pass
      }
    })
    wrapperForm.find('form').at(0).simulate('submit', {preventDefault() {}})
    expect(wrapperForm.find('#labelNumber').at(0).text()).toMatch("at least one number?")
  })
  it('fail check special character', () => {
    let pass = 'agrhaganteng'
    wrapperForm.find('input').at(3).simulate('change', {
      target: {
        name: 'password',
        value: pass
      }
    })
    wrapperForm.find('form').at(0).simulate('submit', {preventDefault() {}})
    expect(wrapperForm.find('#labelSpecial').at(0).text()).toMatch("at least one symbol?")
  })
  it('fail check password length', () => {
    let pass = 'agra'
    wrapperForm.find('input').at(3).simulate('change', {
      target: {
        name: 'password',
        value: pass
      }
    })
    wrapperForm.find('form').at(0).simulate('submit', {preventDefault() {}})
    expect(wrapperForm.find('#labelLength').at(0).text()).toMatch("minimal 5 letter length?")
  })
})


describe('EditForm checked', () => {
  const wrapperEdit = shallow(<EditForm/>, {context:{store}})
  it('editPassword should have form', () => {
    expect(wrapperEdit.find('form')).toBeTruthy()
  })
})

it('renders App correctly', () => {
  let tree = renderer.create( 
  <Provider store={store}>
    <App user={{loginStatus: false}}/>
  </Provider>)
  expect(tree).toMatchSnapshot();
});