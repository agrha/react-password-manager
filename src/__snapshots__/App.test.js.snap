// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <a
      href="/home"
      onClick={[Function]}
    >
      <h1
        className="App-title"
      >
        Welcome To React
      </h1>
    </a>
  </header>
  <div
    className="container"
  >
    <div
      className="flex-frame"
    >
      <form
        onSubmit={[Function]}
      >
        <fieldset>
          <h1>
            <strong>
              Login Form
            </strong>
          </h1>
          <div
            className="form-group"
          >
            <label
              htmlFor="inputUsername"
            >
              Username
            </label>
            <input
              className="form-control"
              id="inputUsername"
              name="username"
              onChange={[Function]}
              placeholder="Enter Your Username"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="inputPass"
            >
              Password
            </label>
            <input
              autoFocus={true}
              className="form-control"
              id="inputPass"
              name="password"
              onChange={[Function]}
              placeholder="Enter Your Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group bold"
          >
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={undefined}
                className="custom-control-input"
                id="checkUpper"
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="checkUpper"
                id="labelUpper"
              >
                at least one uppercase letter?
              </label>
            </div>
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={undefined}
                className="custom-control-input"
                id="checkLower"
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="checkLower"
                id="labelLower"
              >
                at least one lowercase letter? 
              </label>
            </div>
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={undefined}
                className="custom-control-input"
                id="checkSpecial"
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="checkSpecial"
                id="labelSpecial"
              >
                at least one symbol?
              </label>
            </div>
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={undefined}
                className="custom-control-input"
                id="checkNumber"
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="checkNumber"
                id="labelNumber"
              >
                at least one number?
              </label>
            </div>
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={undefined}
                className="custom-control-input"
                id="checkLength"
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="checkLength"
                id="labelLength"
              >
                minimal 5 letter length?
              </label>
            </div>
          </div>
          <button
            className="btn btn-primary button"
            disabled={true}
            type="submit"
          >
            Submit
          </button>
          <a
            href="/register"
            onClick={[Function]}
          >
            <button
              className="btn btn-primary button"
            >
              To Register
            </button>
          </a>
        </fieldset>
      </form>
    </div>
  </div>
</div>
`;
